### PR TITLE
Adding Snapshot Strategy for Moxie

### PR DESCRIPTION
**Pull request by Hrishikesh-Thakkar**
Changes proposed in this pull request:
- Adding a new snapshot strategy for Moxie

### CODE REVIEW

File: src/strategies/moxie/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# moxie
+
+This strategy returns the voting power of users. This takes into consideration
+1. Vesting Contract Moxie Balance
+2. Wallet Moxie Balance
+3. Vesting Contract Fan Token Balance
+4. Wallet Fan Token Balance
+5. Liquidity Pool Contributions
```
Comment (0xsarvesh) [inline src/strategies/moxie/README.md:8]: staked fan tokens for wallet & vesting contracts.

File: src/strategies/moxie/index.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      allAddressesScoreMap[userAddress] += parseFloat(formatUnits(portfolio.unstakedBalance, MOXIE_DECIMALS)) * UNSTAKED_FAN_TOKEN_MULTIPLIER * portfolio.subjectToken.currentPriceInMoxie +
+        parseFloat(formatUnits(portfolio.stakedBalance, MOXIE_DECIMALS)) * STAKED_FAN_TOKEN_MULTIPLIER * portfolio.subjectToken.currentPriceInMoxie;
+    });
+
+    if (next_page == -1) break;
+    next_page++;
+  }
+
+  next_page = 0;
+  const liquidityPoolSubgraphQuery = {
```
Comment (0xsarvesh) [inline src/strategies/moxie/index.ts:102]: can we make parallel calls with portfolio subgraph

File: src/strategies/moxie/index.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+      allAddressesScoreMap[userAddress] += MOXIE_LIQUIDITY_MULTIPLIER * parseFloat(formatUnits(userPool.totalLPAmount, MOXIE_DECIMALS)) *
+        parseFloat(formatUnits(userPool.pool.moxieReserve, MOXIE_DECIMALS)) /
+        parseFloat(formatUnits(userPool.pool.totalSupply, MOXIE_DECIMALS));
+    });
+    if (next_page == -1) break;
+    next_page++;
+  }
+
+  // RPC Call to get balance of Moxie at a block for users
+  const multi = new Multicaller(network, provider, abi, { blockTag });
```
Comment (0xsarvesh) [inline src/strategies/moxie/index.ts]: this can be parallel too

**Review by 0xsarvesh — CHANGES_REQUESTED** [review]

**Review by 0xsarvesh — COMMENTED** [review]
overall looks good, what happen if network call fails?

**Review by sharathkrml — APPROVED** [review]
LGTM

**Pull request merged**